pyscf/mcpdft/tfnal_derivs.py

```diff
@@ -106,8 +106,7 @@ def eval_ot(otfnal, rho, Pi, dderiv=1, weights=None, _unpack_vot=True):
     assert (rho.shape[0] == 2)
     nderiv = rho.shape[1]
     nderiv_Pi = Pi.shape[0]
-    if nderiv > 4:
-        raise NotImplementedError("Translation of meta-GGA functionals")
+
```

Comment:
Add a new `if nderiv > 6:` even though I don't believe a functional would even exist. Or make it an assert just as a sanity check.

Reply:
I haven't added this line, because we won't be passing rho.shape[1] > 6 to this function.

Reply:
it could still be something good to assert as *if* `rho.shape[1] > 6` we have an issue somewhere in the logic of the code (somewhere else).

Reply:
Yes. I can add that assertion.

Reply:
I think we don't have to add this assertion as well, because we are [generating](https://github.com/pyscf/pyscf-forge/blob/6c1acbdcb8a1ac272638221b3e99c38d8fc3492f/pyscf/mcpdft/otfnal.py#L80C1-L81C23) the rho, rho', laplacian and tau using 
[dft.numint.eval_rho2 function](https://github.com/pyscf/pyscf/blob/6878b69129e33432bca2249fdfebfd0075f329f2/pyscf/dft/numint.py#L328), which will won't return rho.shape[1]>6.

Reply:
Yeah, it won't now. But what about later, when other people try to do different things? One should imagine that each function is being written in total isolation and every function that feeds into it is being constantly being edited by a million strangers with a million different objectives.

Reply:
Yah, it is best to indicate *intent* of the function parameters using `asserts` other `if` statements. This makes it easier to debug later down the line, and so that other users don't try and improperly abuse the functions.

Reply:
Sure.

I will add the assertion.

Reply:
Done.